tests/std/tests/GH_004879_span_constructed_by_basic_ispanstream/test.compile.pass.cpp

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// GH-4879 <spanstream>: The span constructed by basic_ispanstream's range constructor may be ill-formed
+
+#include <spanstream>
+
+struct R {
+    char* begin();
+    std::unreachable_sentinel_t end();
+    operator std::span<const char>();
+};
+
+int main() {

```

Comment:
Our compile-only tests traditionally don't have a `main` as a reminder that they never run. We should rename this function to something like `void test()`. (You can pick a less boring name 😸.)